src/motor_node.cc

```diff
@@ -53,49 +57,146 @@ void PID_update_callback(const ubiquity_motor::PIDConfig& config,
     }
 
     if (level == 1) {
-        firmware_params.pid_proportional = config.PID_P;
+        g_firmware_params.pid_proportional = config.PID_P;
     } else if (level == 2) {
-        firmware_params.pid_integral = config.PID_I;
+        g_firmware_params.pid_integral = config.PID_I;
     } else if (level == 4) {
-        firmware_params.pid_derivative = config.PID_D;
+        g_firmware_params.pid_derivative = config.PID_D;
     } else if (level == 8) {
-        firmware_params.pid_denominator = config.PID_C;
+        g_firmware_params.pid_denominator = config.PID_C;
     } else if (level == 16) {
-        firmware_params.pid_moving_buffer_size = config.PID_W;
+        g_firmware_params.pid_moving_buffer_size = config.PID_W;
     } else if (level == 32) {
-        firmware_params.pid_velocity = config.PID_V;
+        g_firmware_params.pid_velocity = config.PID_V;
     } else if (level == 64) {
-        firmware_params.max_pwm = config.MAX_PWM;
+        g_firmware_params.max_pwm = config.MAX_PWM;
     } else {
         ROS_ERROR("Unsupported dynamic_reconfigure level %u", level);
     }
 }
 
+//  initMcbParameters()
+//
+//  Setup MCB parameters that are from host level options or settings
+//
+void initMcbParameters(std::unique_ptr<MotorHardware> &robot )
```

Comment:
I am thinking that this might be better to put in MotorHardware directly, there is no need to clutter the node stuff with it.

Also if you keep it out here, there is no need to pass a unique_ptr by ref, you should be passing a MotorHardware by ref into this function.

`void initMcbParameters(MotorHardware& robot)`

Which can be called in main by `initMcbParameters(*robot)`

Reply:
Well here is my take on why it belongs in motor_node.cc:
I see motor_node.cc the 'mother function' and in fact the node itself.    This is where the ROS parameters should be recognized.  I see motor_hardware.cc as the 'bunch of hardware specific worker routines.
Because it is in a function now I do not see this as 'cluttering'   in fact it was more cluttered before with a lot of this right in the main for the node.    So I see motor_node.cc as the place where high level node parameters are passed to the node and make their way to lower level.   I also see motor_node.cc as the place where decisions that vary the node operation should mostly happen.   It is  'the app' is my thought.  All else (hardware, messages, serial) are the things that are envolked by the main 'app'.     I hope you agree.